src/api/getBaladyiatsForWilaya.js

```diff
@@ -0,0 +1,23 @@
+const projectWilaya = require("../utils/projections/wilayaProjection");
+
+const getBaladyiatsForWilaya = data =>
+/**
+   * Takes a wilaya code (mattricule) and returns array of Baladiyates of wilaya.
+   *
+   * @example Get Baladiyats list of Alger (code: 16)
+   *
+   *
+   * getBaladyiatsForWilaya (16)
+   *
+   * @param { Number } mattricule wilaya code (mattricule)
+   * @param {String[]} projection a list of Baladyia object attributes to keep
+   * @returns { Object[] }
+   */
+
+  (mattricule, projections) => {
+    const baladyiats = [].concat(...data.find((wilaya) => wilaya.mattricule === mattricule).dairats.map((daira) => daira.baladyiats));
+    return projectWilaya(baladyiats, projections);
```

Comment:
I'd do it like this, probably in more lines but it could be easily readable.

```suggestion
  (mattricule, projections) => {
    const wilaya = data.find(w => w.mattricule === mattricule);

    if (wilaya) {
      const baladyiats = wilaya.dairats.reduce((acc, daira) => [...acc, ...daira.baladyiats], []);
      return projectWilaya(baladyiats, projections);
    }

    return null;
  };
```

Reply:
Just to make sure about one thing : 
By checking if the entered Wilaya exists only ( without checking dairats array )  with `if (wilaya)` you are assuming once i find a wilaya i'm sure that it has Daira and Baladyia. but why checking wilaya if i'm sure that the entered wilaya mattricule is between : [ 1 - 48 ].

Reply:
> By checking if the entered Wilaya exists only ( without checking dairats array ) with if (wilaya) you are assuming once i find a wilaya i'm sure that it has Daira and Baladyia.

Yes, every wilaya has Dairats array, and each Daira has Baladyiats.

> but why checking wilaya if i'm sure that even the entered wilaya mattricule is between : [ 1 - 48 ] ?

for now, it's just to make it safe and return either null or the data array. But sure, we may add a check for that inside each function taking the wilaya code in parameters, and throw an error if it's invalid/out-of-range.